File: 538.ConvertBSTtoGreaterTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function(root) {
    let sum = 0;
    const stack = [];
    let current = root;
    while(true) {
        if (current) {
            stack.push(current);
            current = current.right;
        } else if (stack.length) {
            const node = stack.pop();
            sum += node.val;
            node.val = sum;
            current = node.left;
        } else {
            break;
        }
    }
    return root;
};

File: Golang/Tree/538.Convert-bst-to-greater-tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func convertBST(root *TreeNode) *TreeNode {
	previous := 0
	var recursive func(node *TreeNode)
	recursive = func(node *TreeNode) {
		if node != nil {
			recursive(node.Right)
			node.Val += previous
			previous = node.Val
			recursive(node.Left)
		}
	}
	recursive(root)
	return root
}
